netbox/users/tests/test_api.py

```diff
@@ -111,10 +111,30 @@ def test_password_validation_enforced(self):
         self.assertEqual(response.status_code, 400)
 
         # Password long enough
-        data['password'] = 'foobar123'
+        data['password'] = 'FooBar123'
         response = self.client.post(url, data, format='json', **self.header)
         self.assertEqual(response.status_code, 201)
 
+        # Password no number
+        data['password'] = 'foobarFoo'
+        response = self.client.post(url, data, format='json', **self.header)
+        self.assertEqual(response.status_code, 400)
+
+        # Password no letter
+        data['password'] = '123456789012'
+        response = self.client.post(url, data, format='json', **self.header)
+        self.assertEqual(response.status_code, 400)
+
+        # Password no uppercase
+        data['password'] = 'foobarfoo1'
+        response = self.client.post(url, data, format='json', **self.header)
+        self.assertEqual(response.status_code, 400)
+
+        # Password no lowercase
+        data['password'] = 'FOOBARFOO1'
+        response = self.client.post(url, data, format='json', **self.header)
+        self.assertEqual(response.status_code, 400)
```

Comment:
This set of logic tests is duplicated for the UI and API. Can we move it to a single test module dedicated to testing password validation on the User model itself?

Reply:
IMHO: I think it needs to be here as https://github.com/netbox-community/netbox/blob/develop/netbox/users/forms/model_forms.py#L231 specifically calls validate_password so these checks make sure it is getting called appropriately in the form.